cli/src/migrate.js

```diff
@@ -0,0 +1,354 @@
+    metavar: 'host:port',
+    default: undefined,
+    help: 'Host and port of the RethinkDB server to connect to.',
+  });
+
+  parser.addArgument([ '--rdb-user' ], {
+    default: 'admin',
+    metavar: 'USER',
+    help: 'RethinkDB User',
+  });
+
+  parser.addArgument([ '--rdb-password' ], {
+    default: undefined,
+    metavar: 'PASSWORD',
+    help: 'RethinkDB Password',
+  });
+
+  parser.addArgument([ '--start-rethinkdb' ], {
+    action: 'storeTrue',
+    help: 'Start up a RethinkDB server in the current directory'
+  });
+
+  parser.addArgument([ '--config' ], {
+    default: '.hz/config.toml',
+    help: 'Path to the config file to use, defaults to ".hz/config.toml".',
+  });
+
+  parser.addArgument([ '--skip-backup' ], {
+    action: 'storeTrue',
+    help: 'Whether to perform a backup of rethinkdb_data before migrating',
+  });
+
+  const parsed = parser.parseArgs(cmdArgs);
+  const confOptions = config.read_from_config_file(parsed.project_path, parsed.config);
+  // Pull out the relevant settings from the config file
+  const options = {
+    project_path: parsed.project_path || '.',
+    project_name: parsed.project_name || confOptions.project_name,
+    rdb_host: parsed.rdb_host || confOptions.rdb_host || 'localhost',
+    rdb_port: parsed.rdb_port || confOptions.rdb_port || 28015,
+    rdb_user: parsed.rdb_user || confOptions.rdb_user || 'admin',
+    rdb_password: parsed.rdb_password || confOptions.rdb_password || '',
+    skip_backup: parsed.skip_backup || false,
+    start_rethinkdb: parsed.start_rethinkdb || false,
+  };
+  // sets rdb_host and rdb_port from connect if necessary
+  if (parsed.connect) {
+    config.parse_connect(parsed.connect, options);
+  }
+
+  if (options.project_name == null) {
+    throw new Error('project_name is null');
+  }
+  return options;
+}
+
+function setup() {
+  // Start rethinkdb server if necessary
+  // Connect to whatever rethinkdb server we're using
+  white('Setup');
+  return Promise.resolve().then(() => {
+    if (this.options.project_path && this.options.project_path !== '.') {
+      green(` ├── Changing to directory ${this.options.project_path}`);
+      change_to_project_dir(this.options.project_path);
+    }
+  }).then(() => {
+    // start rethinkdb server if necessary
+    if (this.options.start_rethinkdb) {
+      green(' ├── Starting RethinkDB server');
+      return start_rdb_server({ quiet: true }).then((server) => {
+        this.rdb_server = server;
+        this.options.rdb_host = 'localhost';
+        this.options.rdb_port = server.driver_port;
+      }).then(() => Promise.delay(2000));
+    } else {
+      return undefined;
+    }
+  }).then(() => {
+    green(' ├── Connecting to RethinkDB');
+    return r.connect({
+      host: this.options.rdb_host,
+      port: this.options.rdb_port,
+      user: this.options.rdb_user,
+      password: this.options.rdb_password,
+    });
+  }).then((conn) => {
+    green(' └── Successfully connected');
```

Comment:
Specifically, we need `hz_collections` to be ready, I don't think we touch the data in any of the others.